File: 242.ValidAnagram/solution.py
"""
Time complexity => O(s+t)
Space complexity => O(s+t)
"""

from collections import Counter


def is_anagram(s: str, t: str) -> bool:
    count_s = Counter(s)
    count_t = Counter(t)
    print(count_s == count_t)


s = "hello"
t = "hello"

is_anagram(s, t)

File: 242.ValidAnagram/solution.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */

function isAnagram(s, t) {
  if (s.length !== t.length) return false;

  const count = {};

  for (let i = 0; i < s.length; i++) {
    if (!count[s[i]]) count[s[i]] = 0;
    if (!count[t[i]]) count[t[i]] = 0;
    count[s[i]]++;
    count[t[i]]--;
  }
  for (let c in count) {
    if (count[c] !== 0) return false;
  }
  return true;
}
